addons/l10n_in_edi/models/account_move.py

```diff
@@ -27,6 +28,19 @@ def _compute_l10n_in_edi_show_cancel(self):
                 and i.state in ("sent", "to_cancel", "cancelled")
             ))
 
+    @api.depends('state', 'journal_id', 'invoice_line_ids')
+    def _compute_l10n_in_edi_show_negative_line_info(self):
+        indian_invoices = self.filtered(lambda m: m.country_code == 'IN' and m.journal_id.type == 'sale')
+        (self - indian_invoices).l10n_in_edi_show_negative_line_info = False
+        for invoice in indian_invoices:
+            show_negative_line_info = False
+            if invoice.invoice_line_ids.filtered(lambda l: l.price_total < 0):
+                if invoice.state == 'draft' and invoice.journal_id.compatible_edi_ids.filtered(lambda i: i.code == "in_einvoice_1_03" and i._get_move_applicability(invoice)):
```

Comment:
Why make the condition that complicated?

Reply:
it's simple.
if E-invoice is applicable and has a negative line for the `draft`.
if E-invoice is `to_send` and has a negative line for the `posted`.

Reply:
but look complicated :wink: